src/vm/arm64/asmhelpers.asm

```diff
@@ -1121,12 +1121,12 @@ MainLoop
         cmp     x9, #0
         beq     Fail
 
-        ldr     x9,  [x9, #0]
-        cmp     x9, x13           ; compare our MT with the one in the ResolveCacheElem
+        ldr     x16, [x9, #0]
```

Comment:
If you have a second register available you can use a ldp instruction:

ldp   x16,x17, [x9]

you don't need the ", #0" as that is implied by default.